scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -1228,6 +1237,39 @@ public final void handshakeFailed(Throwable cause) {
 		SecretUtil.destroy(this);
 	}
 
+	/**
+	 * Test, if handshake was started in probing mode.
+	 * 
+	 * Usually a client handshake removes the session from the connection store
+	 * with the start. Probing removes the session only with the first data
+	 * received back.
+	 * 
```

Comment:
Maybe adding the link to your https://github.com/eclipse/californium/pull/1157#issuecomment-561223737 here too